specs/File Classification Spec.md

```diff
@@ -0,0 +1,59 @@
+# **PowerToys File Classification Spec**
+
+![MS PowerToys](https://hothardware.com/ContentImages/NewsItem/48038/content/Microsoft_PowerToys.jpg "PowerToys")
+## Users can quickly rename and group files.
+### Authors: Benjamin Leverette and Prudence Phillips
+### Spec Status: Draft
+## 1. Overview
+
+**1.1. Elevator Pitch / Narrative** 
+
+David has recently uploaded thousands of pictures from his camera to his Surface Laptop.  Unfortunately, the images all have generic names such as 'IMG_141' instead of labels that he can identify.  David downloads the File Classification PowerToy that allows him to create new folders with files that share a label, rename a group of files with a label, or change the name or portion of a name in a group of files to another name.
+
+**1.2. Customers**
+
+Like all utilities from PowerToys, the File Classification feature is for power users and developers who are looking to tune and streamline their Windows experience for greater productivity.
+  
+**1.3. Problem Statement and Supporting Customer Insights**
+
+Power users need a better way to organize files, from renaming files to creating new folders for similar files.  Our PowerToys Consumer Survey received feedback validating the usefulness of a feature that provides such functionality.
+
+**1.4. Existing Solutions or Expectations**
+
+Users currently have to highlight a group of files and right click to rename them.  Users must manually create a folder, move files to that folder, and rename them.  There are third-party resources that allow users to rename files similarly.
+
+We expect users to install and enable PowerToys for Windows in order to access the File Classification utility.
+
+**1.5. Goals/Non-Goals**
+
+Design and develop a feature that can rename and group files within an 8 week period.
+
+## 2. Definition of Success
+
+**2.1. Expected Impact: Customer, and Technology Outcomes, Experiments + Measures**
+
+Our PowerToys Consumer Survey received an abundant amount of participation and feedback from a community of passionate power users.  This feature will give them the ability to rename, group and organize files in a way that makes their virtual library of files more organized and efficient.  As interns, we would have 8 weeks to complete the project.
+
+## 3. Requirements
+
+**3.1.	Functional Requirements**
+
+- Users must select a group of files by checking each file or using Shift + directional keys.  They must then use a shortcut to open the File Classification utility, which provides three options:
+  - Automates folder creation and movement of user-selected files into that folder after receiving a label determined by the user.
+  - Renames all user-selected files after receiving a label determined by the user.
+  - Identifies a string of characters in the names of user-selected files and changes it to a string determined by the user.
```

Comment:
How are you identifying this string? Can I have wildcard characters in it? Or are there any other options I can use when identifying these strings?
For example in bash I can specify which files I'd like to rename using wild cards, etc. 
![image](https://user-images.githubusercontent.com/42221804/58653800-58206200-82cb-11e9-9f95-984975541df1.png)

Reply:
The files that we are checking are already selected by the user before the utility is opened.  This provides the pool of file names that are checked.  The user determines the characters we look for as well as what we will replace those characters with each time we find the characters in the file names.

Thus, they can type in a series of characters of any form.  If it is not in any of the files' names then nothing is replaced.

I'll see to it that the spec depicts this more clearly.  Thanks!